### PR TITLE
Cog1 Catering Maint Expansion

### PR DESCRIPTION
**Pull request by Nexusuxen**
## About the PR 

IMAGES AT THE BOTTOM OF THIS SECTION

UPDATE: The meat vending machine has been removed, and this will also remove the new Random Room that was recently added to this area.

Adds new back rooms to Catering Maintenance on Cogmap1, and gives the left and right entrances to the maintenance shaft bartender and chef access (though not to the construction area). Expands the Catering Router by a couple tiles, putting tables in the new tiles with a produce satchel and wrapping paper and scissors.

The 'Catering Backroom' - the larger boarded up room - contains the following additions:

- A modified version of Warcrimes' Meat4Cash vendor, renamed to FreshFlesh (the original vendor has been made a child of a new abstract obj, which is also the parent of the 'station-friendly' variant). The current stock is as follows (does not show the hidden 5 pieces of human meat, sold for 1k credits each)
image
Modification was made due to concerns that 42 pieces of meat available to the chef (and really anyone with maint access) for very cheap broke some balancing factors and trivialized the time-honored tradition of harvesting monkey meat.
- A broken gibber that, while useless, may be deconstructed and removed. People noted having an actual gibber in a cameraless maint backroom on-station would trivialize body disposal just a bit too much. The decorative variant is also more thematic.
- A Shitty Grill, for the wandering staffie to make their own backwater kitchen, and for the chef to grill things on
- Assorted catering equipment, including plates, 3 folding tables, beer in a box, a produce satchel, and a spare meatspike. Also random weird stuff, including a trowel, cardboard tube, monkey plushie, hand labeler, and the Swedish Fisk has been moved from the kitchen to this back room.
- Some random objects scattered across the floor, including a fancy pen, clipboard, briefcase, and NT beret, which clearly have no connection to one another, or so says the previous chef that worked here.
- NO CAMERAS, giving this general region a decent place to hide from the prying eyes of the AI.

A secondary maint nook has also been added, accessed via false wall, just to the right of the Catering Backroom. It includes:
- A "sleeping bag"
- A chair
- Assorted dirt and debris
- A trash pickle (delicious!)
- A ??? pill bottle
- A dried up joint
- A junk Spawner
- A great view ~into the solar array~ into space

In-game view (with fullbright enabled)
image

In-game view (with fullbright disabled)
image

Map editor view
image

## Why's this needed? 

There aren't really that many places to hide from the AI in the vicinity of Catering - just about everywhere has camera coverage, save for small parts of the chapel. This helps to rectify that issue.

More maint backrooms with weird stories also adds some life to maint, and makes it feel more lived-in and used.

Provides genuinely useful things, especially to the chef, hopefully without causing imbalance issues. The meat vendor hopefully provides the chef a way to get rarer meats if botany is not cooperating, albeit at a ludicrous price to limit how much can be purchased. The grill and tables are also nice if a chef wants to use them - the tables in particular can be used to set up food displays in other departments, and the chef normally doesn't have access to a general fabricator, and as such can't get any normally.

Consistency with the general pattern of having crew being able to access their own maintenance shafts, even if they normally don't have maint access. (The bartender and chef were already able to access this area anyways, so giving them access to the other 2 entrances just makes sense)

## Changelog

```changelog
(u)Nexusuxen
(*)Catering maintenance on Cogmap1 has been expanded. Check it out!
```

### CODE REVIEW

File: code/WorkInProgress/warcrimes.dm [hunk 1]
```diff
@@ -81,14 +82,33 @@ var/fartcount = 0
...
 		..()
-		product_list += new/datum/data/vending_product(/obj/item/reagent_containers/food/snacks/ingredient/meat/mysterymeat, 10, cost=PAY_UNTRAINED/4)
-		product_list += new/datum/data/vending_product(/obj/item/reagent_containers/food/snacks/ingredient/meat/monkeymeat, 10, cost=PAY_UNTRAINED/5)
-		product_list += new/datum/data/vending_product(/obj/item/reagent_containers/food/snacks/ingredient/meat/synthmeat, 20, cost=PAY_UNTRAINED/6)
 
-		product_list += new/datum/data/vending_product(/obj/item/reagent_containers/food/snacks/ingredient/meat/humanmeat, 2, cost=PAY_UNTRAINED, hidden=1)
+/obj/machinery/vending/meat/prefab_grill
+	name = "Meat4cash"
+	desc = "An exotic meat vendor."
+	create_products()
```
Comment (ZeWaka) [inline code/WorkInProgress/warcrimes.dm:89]:
```suggestion
	desc = "An exotic meat vendor."

	create_products()
```

**Review by ZeWaka — APPROVED** [review]

**Review by flrsh — CHANGES_REQUESTED** [review]
Having a meat vending machine seems pretty out of theme with the rest of the map. Would be much better suited to some space derelict or prefab.

**Comment by Nexusuxen** [comment]
> Having a meat vending machine seems pretty out of theme with the rest of the map. Would be much better suited to some space derelict or prefab.

While I disagree, if need be I can remove it. I implemented it since it was thematic to the sort of meat processing equipment the room has, given the 'spare' gibber and monkey spike. It was also intended to provide some rarer kinds of meat that are harder to get if botany isn't cooperating. If another dev/maintainer/admin could weigh in, or multiple, I'd really appreciate that. If I don't get any significant feedback regarding it the next few days I'll remove it.

**Comment by Nexusuxen** [comment]
I have not been maintaining this due to the combination of:
- Semi-frequent updates to Cog1 requiring consistent merge conflict resolution
- Lack of response to requested re-reviews and lack of further reviews

I am still interested in keeping this open, so I'll try to resolve the merge conflicts later today to avoid the conflicts becoming too severe.

**Comment by ZeWaka** [comment]
@Nexusuxen 
> I have not been maintaining this due to the combination of:
> 
> * Semi-frequent updates to Cog1 requiring consistent merge conflict resolution
> * Lack of response to requested re-reviews and lack of further reviews
> 
> I am still interested in keeping this open, so I'll try to resolve the merge conflicts later today to avoid the conflicts becoming too severe.

I'd like to merge this, with the redaction of the vending machine for now.

**Comment by Nexusuxen** [comment]
Merge conflict resolved & vendor removed.

NOTE: In the time that this PR has been gathering dust, a Random Room was added here, and as such this PR will necessitate its removal.

**Review by ZeWaka — APPROVED** [review]
We can re-add the random room elsewhere or back there later.

**Comment by TTerc** [comment]
!merge_upstream

**Review by TTerc — APPROVED** [review]

**Pull request merged**